### PR TITLE
787 pspm_downsample

### LINKED ISSUE #787: Problem with downsampling missing values in `pspm_dcm`.
In what is currently line 404 of `pspm_dcm`, the function `downsample` is called with a potentially non-integer sample rate, which will throw an error. The function `resample` (used in `pspm_prepdata`) would work here, but only if the signal processing toolbox is installed.

Suggested workaround:
- `pspm_downsample` becomes the main function for handling downsampling, and integrates lines 154-178 of `pspm_prepdata`
- `pspm_prepdata` and `pspm_dcm` call `pspm_downsample` for downsampling. 
- create tests for `pspm_downsample` 

_Originally posted by @dominikbach in https://github.com/bachlab/PsPM/discussions/719#discussioncomment-10870203_

### PR DESCRIPTION
**Pull request by 4gwe**
Fixes  #787 .

Changes proposed in this pull request:
- pspm_downsample got the resample feature from (pspm_prepdata)
- pspm_prepdata uses  pspm_downsample
-  pspm_downsample uses pspm_dcm (implementation in progress).
- created pspm_downsample_test

### CODE REVIEW

File: src/pspm_downsample.m [hunk 1]
```diff
@@ -23,10 +24,42 @@
...
+%   warning('Frequency ratio must be integer.'); return
 end
 %% 3 Performing downsampling
-data = data(freqratio:freqratio:end);
-sts = 1;
+
+freqratio = sr/sr_down;
+
+
+if freqratio == ceil(freqratio) % NB isinteger might not work for some values
```
Comment (dominikbach) [inline src/pspm_downsample.m:35]: I'd suggest exchanging the first and second case, i.e. start with `if settings.signal`. This way, we can be sure that an in-built function handles most of the cases, even when frequency ratio is integer.

File: src/pspm_downsample.m [hunk 2]
```diff
@@ -23,10 +24,42 @@
 %% 2 Check input arguments
 if nargin < 2
   warning('Not enough input arguments.'); return
-elseif floor(freqratio) ~= freqratio
-  warning('Frequency ratio must be integer.'); return
+% elseif floor(freqratio) ~= freqratio
```
Comment (dominikbach) [inline src/pspm_downsample.m:27]: Please delete the commented out lines.

File: src/pspm_downsample.m [hunk 3]
```diff
@@ -1,9 +1,9 @@
-function [sts, data] = pspm_downsample(data, freqratio)
+function [sts, data, newsr] = pspm_downsample(data, sr,sr_down)
 % ● Description
```
Comment (dominikbach) [inline src/pspm_downsample.m:2]: Please change help text. This function is now used more generally, not just for users without signal processing toolbox.

**Review by dominikbach — CHANGES_REQUESTED** [review]

File: src/pspm_downsample.m (hunk 1 above)
Comment (4gwe) [reply src/pspm_downsample.m:35]: But will the resampled data not be a bit off (in the case of integer ratios)?

**Review by dominikbach — APPROVED** [review]

**Pull request merged**